compiler/src/lnntoamm/Expr.ts

```diff
@@ -92,28 +93,202 @@ export default abstract class Expr {
   }
 
   static fromAssignablesAst(ast: LPNode, metadata: MetaData): [Stmt[], Expr] {
+    const asts = ast.getAll();
     // break it up so that we're only working on one base assignable list or operator at a time.
-    let operated = ast.getAll().map(work => {
+    let operated: Array<[Stmt[], Expr] | Operator[]> = asts.map(work => {
       work = work.get('withoperators');
       if (work.has('baseassignablelist')) {
         return Expr.fromBaseassignablelist(work.get('baseassignablelist'), metadata);
       } else if (work.has('operators')) {
-        return TODO('operators');
+        // TODO: this won't work with operators associated with interfaces.
+        // Will have to iterate through all of the interfaces in-scope and collect
+        // the applicable types as well
+        const op = work.get('operators').t.trim();
+        let operators = metadata.scope.get(op) as Operator[];
+        if (operators === null) {
+          console.log(metadata.scope);
+          throw new Error(`can't find operator ${op}`);
+        } else if (!isOpArray(operators)) {
+          // sanity check
+          throw new Error(`somehow ${op} isn't an operator?`);
+        }
+        return operators;
       } else {
         throw new Error(`unexpected assignable ast: ${work}`);
       }
     });
     if (operated.length === 0) {
       throw new Error(`no expressions generated for ast: ${ast}`);
     } else if (operated.length === 1) {
-      if (operated[0][0] instanceof Array) {
-        return operated[0] as [Stmt[], Expr];
-      } else {
+      if (isOpArray(operated)) {
         throw new Error(`variables can't be assigned to operators`);
       }
-    } else {
-      return TODO('operators');
+      return operated[0] as [Stmt[], Expr];
+    }
+    // now we have to resolve operators - start by filtering out operators if they
+    // are in a position that must be prefix or infix
+    // since there are no suffix operators, this is relatively easy - operators
+    // immediately following an expression must be infix, while all others must be
```

Comment:
```ln
const x = 5 + ++y;
```

(We don't have an increment operator, but we do need to make sure this kind of thing works.)

Maybe a more likely example:

```ln
const shouldRun = inValidState && !inUse;
```

Reply:
in these cases, the prefix operator is higher precedence so it gets resolved first. You're right that that'll trigger an operator ambiguity error though - I'll fix it